### ISSUE TITLE
What the paramaters for training JHMDB-21

### ISSUE THREAD

**Issue by jiaozizhao** [gurkirt/realtime-action-detection#25]
Hi, I am trying to train models on JHMDB-21 using your code. I only modify the ratios of each class. But the results are very bad. For the first split of JHMDB, the frame-map@0.5 is only 46%. Do you have any suggestions for training JHMDB?

**Comment by gurkirt** [comment]
46 seems okay, max you going to get is about 48-51%. But don't go on frames mAP on jhmdb21, check video mAP. It is a small dataset so frame-mAP is usually unstable. You can see the original SSD settings here https://pdfs.semanticscholar.org/50be/f2075a6f50a2525c3166a14ad413b7d38a0e.pdf.

I think the initial learning rate of 0.0005 and then drop it after 2.5k or 5k iteration should do it on rgb stream bu ton flow

**Comment by jiaozizhao** [comment]
Thanks for quick reply. I will try.

**Comment by jiaozizhao** [comment]
Hi, Gurkirt. I tried to train JHMDB only rgb images again as you said. The frame-map is about 46%. However, the video-mAP @0.2 is only 46% which is very far from 60.8% reported in your paper. As I need to compare with your method, I really need to repeat your results. Could you give me some suggestions training on JHMDB, please? Or can you release some configuration files for training JHMDB?

**Comment by gurkirt** [comment]
I am not sure if you are using correct protocol and annotation fo rJHMDB-21. Try tube building and evaluation protocol from https://bitbucket.org/sahasuman/bmvc2016_code/src/master/

**Comment by gurkirt** [comment]
Also, RGB is a small contributor on JHMDB compared to optical flow.

**Comment by jiaozizhao** [comment]
I used the codes under the folder 'online_tubes' in this project to evaluate JHMDB. I know you just gave ucf101 as an example in this project. But I think the evaluation method should be same for JHMDB, right? So I modified it to test JHMDB. Or I have to change to use the evaluation from https://bitbucket.org/sahasuman/bmvc2016_code/src/master/? Is it different evaluation from this project?

**Comment by gurkirt** [comment]
Should work. Just take care the fact that JHMDB videos are trimmed, unlike in ucf where videos are untrimmed.
Good luck

**Comment by jiaozizhao** [comment]
Hi, I am sorry to disturb you again. But I am still not clear about what I should take care about trimmed videos JHMDB for the evaluation. The evaluation should be same with the untrimmed videos UCF101, right?  I used the same evaluation for ucf101 and JHMDB, but the video-mAP for UCF101 is same with you reported in the paper and for JHMDB is much lower than the paper. I am really confused.

**Comment by jiaozizhao** [comment]
Hi, Gurkirt,
Do you mean that when I test JHMDB, I should not use the code 'path_smoother.m' for temporal trimming, but only use the action paths after the first pass dynamic programming for evaluating?

**Comment by gurkirt** [comment]
Yes, you are right.
